### PR TITLE
Show key fpr from signature when signature check fails (bsc#1187224) 

### PR DESCRIPTION
**Pull request by bzeller**
Rpm by default only shows the short key ID when checking the signature
of a package fails. This patch reads the signatures from the RPM headers
and replaces she short IDs with the key fingerprints fetched from the signatures.

### CODE REVIEW

File: zypp/target/rpm/RpmDb.cc [hunk 1]
```diff
@@ -1239,6 +1240,44 @@ namespace
...
+      addFprs( RPMTAG_SIGPGP );
+      addFprs( RPMTAG_RSAHEADER );
+      addFprs( RPMTAG_DSAHEADER );
+
+      for ( auto &detail : detail_r ) {
+        auto &line = detail.second;
+
+        // replace the short key ID with the long ones parsed from the header
+        for ( const auto &fpr : fprs ) {
+          str::replaceAll( line, fpr.first, fpr.second );
```
Comment (mlandres) [inline zypp/target/rpm/RpmDb.cc:1276]: As Michael said, rpm may report the full FPR in the future. 
Reading the FPR from the header and replacing the last 8 byte of the reported ID (L1261) must not happen if the reported id is longer than 8.

**Review by mlandres — CHANGES_REQUESTED** [review]

File: zypp/target/rpm/RpmDb.cc (hunk 1 above)
Comment (bzeller) [reply zypp/target/rpm/RpmDb.cc]: Done

File: zypp/target/rpm/RpmDb.cc [hunk 2]
```diff
@@ -1239,6 +1240,54 @@ namespace
...
+
+          if ( !didReadHeader ) {
+            didReadHeader = true;
+
+            // Get signature info from the package header, RPM always prints only the 8 byte ID
+            auto header = RpmHeader::readPackage( path_r, RpmHeader::NOVERIFY );
+            if ( !header ) {
+              ERR << "Failed to read package" << std::endl;
+              return RpmDb::CHK_ERROR;
+            }
```
Comment (mlandres) [inline zypp/target/rpm/RpmDb.cc:1262]: If the header is for whatever reason not readable, I'd prefer to continue with the original workflow without replacing the FPR. Just leave the loop and return the computed value (ret).

**Review by mlandres — CHANGES_REQUESTED** [review]

File: zypp/target/rpm/RpmDb.cc (hunk 2 above)
Comment (bzeller) [reply zypp/target/rpm/RpmDb.cc:1279]: ok

**Review by mlandres — APPROVED** [review]

**Pull request merged**